flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/FieldAggregator.java

```diff
@@ -67,8 +67,21 @@ static FieldAggregator createFieldAggregator(
                             "Use unsupported aggregation or spell aggregate function incorrectly!");
             }
         }
+
+        if (ignoreRetract) {
+            fieldAggregator = new FieldIgnoreRetractAgg(fieldAggregator);
+        }
         return fieldAggregator;
     }
 
     abstract Object agg(Object accumulator, Object inputField);
+
+    Object retract(Object accumulator, Object retractField) {
+        throw new UnsupportedOperationException(
+                String.format(
+                        "Aggregate function %s dose not support retraction,"
```

Comment:
Better to use function name instead of the aggregator class name.

Reply:
BTW, we could introduce the `name()` in `FieldAggregator` to return the name of the aggregation function.